### PR TITLE
v1.1 changes: crash reporting metrics integration

### PR DESCRIPTION
**Pull request by MoumitaM**
## Description of the change

> This PR contains the changes specific to the v1.1 version for crash reporting metrics implementation.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/537)

### CODE REVIEW

File: analytics.js [hunk 1]
```diff
@@ -314,11 +322,11 @@ class Analytics {
...
       )
     ) {
-        // Integrations are ready
-        // set clientIntegrationsReady to be true
-        object.clientIntegrationsReady = true;
-        // Execute the callbacks if any
-        object.executeReadyCallback();
+      // Integrations are ready
+      // set clientIntegrationsReady to be true
+      object.clientIntegrationsReady = true;
```
Comment (saikumarrs) [inline analytics.js:327]: Where is this defined? Seems like the changes from v1 have not been merged properly.

File: utils/constants.js [hunk 2]
```diff
@@ -74,7 +74,7 @@ const MAX_WAIT_FOR_INTEGRATION_LOAD = 10000;
 const INTEGRATION_LOAD_CHECK_INTERVAL = 1000;
 const INTG_SUFFIX = "_RS";
 const POLYFILL_URL =
-  "https://polyfill.io/v3/polyfill.min.js?features=Array.prototype.find%2CArray.prototype.includes%2CPromise%2CString.prototype.endsWith%2CString.prototype.includes%2CString.prototype.startsWith%2CObject.entries";
+  "https://polyfill.io/v3/polyfill.min.js?features=Array.prototype.find%2CArray.prototype.includes%2CPromise%2CString.prototype.endsWith%2CString.prototype.includes%2CString.prototype.startsWith%2CObject.entries%2CObject.values";
```
Comment (saikumarrs) [inline utils/constants.js:77]: What about `Object.keys`?

Comment (MoumitaM) [reply utils/constants.js:77]: Are you suggesting using `Object.keys`?

Comment (saikumarrs) [reply utils/constants.js:77]: nm...it's supported in all browsers.

File: analytics.js (hunk 1 above)
Comment (saikumarrs) [reply analytics.js:327]: Why is it not showing in the diff then?

Comment (saikumarrs) [reply analytics.js:327]: nm...the diff is broken.

File: metrics/error-report/Bugsnag.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+const API_KEY = "{{RS_BUGSNAG_API_KEY}}";
+
+// Errors only from Below SDKs are allowed to reach Bugsnag
+const SDK_FILE_NAMES = [
+  "browser.js",
+  "rudder-analytics.min.js",
+  "rudder-analytics-staging.min.js",
+  "rudder-analytics.js",
+  ...Object.values(configToIntNames).map((e) => `${e}.min.js`),
+  ...Object.values(configToIntNames).map((e) => `${e}-staging.min.js`),
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply metrics/error-report/Bugsnag.js]: done

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).
 
 ### Merged
 
-- Align Ready method's behaviour   [`#522`](https://github.com/rudderlabs/rudder-sdk-js/pull/522)
+- Align Ready method's behaviour [`#522`](https://github.com/rudderlabs/rudder-sdk-js/pull/522)
```
Comment (saikumarrs) [inline CHANGELOG.md:14]: Let's revert the changes from this file.

File: buildspec.staging.yaml [hunk 5]
```diff
@@ -6,21 +6,19 @@ phases:
...
       - npm run buildProdBrowser
       - node -r esm integrationBuildScript.js
-      - sed 's|//# sourceMappingURL=rudder-analytics.min.js.map||' dist/rudder-analytics.min.js  > dist/prod.js
-      - mv dist/prod.js dist/rudder-analytics.min.js
+      - sed -i -e 's|rudder-analytics.min.js.map|rudder-analytics-staging.min.js.map|' -e 's|{{RS_BUGSNAG_API_KEY}}|'$RS_BUGSNAG_API_KEY'|' dist/rudder-analytics.min.js
       - aws s3 cp dist/rudder-analytics.min.js s3://$S3_BUCKET_NAME/v1.1/rudder-analytics-staging.min.js --cache-control max-age=3600 --acl public-read
+      - aws s3 cp dist/rudder-analytics.min.js.map s3://$S3_BUCKET_NAME/v1.1/rudder-analytics-staging.min.js.map --cache-control max-age=3600 --acl public-read
       - mkdir dist/temp
       - cp dist/integrations/* dist/temp
       - for file in dist/temp/* ; do old_name=$(basename "$file"); new_name=$(echo $old_name | awk '{split($0, tmp, "."); print sprintf("%s-staging.%s.%s", tmp[1], tmp[2], tmp[3])}'); mv "dist/temp/$old_name" "dist/temp/$new_name"; done
```
Comment (saikumarrs) [inline buildspec.staging.yaml:18]: We need to change the `soruceMappingURL` value inside the integration SDKs.

File: package.json [hunk 6]
```diff
@@ -8,7 +8,7 @@
       "path": "dist/rudder-analytics.min.js",
       "import": "{ createComponent }",
       "gzip": true,
-      "limit": "38 kB"
+      "limit": "38.5 kB"
```
Comment (saikumarrs) [inline package.json:11]: Please check the final size and update this value.

Comment (MoumitaM) [reply package.json]: Addressed

File: buildspec.staging.yaml (hunk 5 above)
Comment (MoumitaM) [reply buildspec.staging.yaml:19]: Addressed

File: CHANGELOG.md (hunk 4 above)
Comment (MoumitaM) [reply CHANGELOG.md]: Addressed

**Review by saikumarrs — DISMISSED** [review]

**Review by saikumarrs — DISMISSED** [review]

**Review by saikumarrs — APPROVED** [review]
